### PR TITLE
DUMS-79 Mutations whenever a card is clicked.

### PR DESCRIPTION
**Pull request by CurtisUnderwood**
### Description

List of proposed changes:

- Added mutation to update server-side data whenever user changes their card selection

### CODE REVIEW

File: pages/room/[id].js [hunk 1]
```diff
@@ -32,11 +34,35 @@ export default function Room(props) {
...
+    card: null,
+    room: props.roomID,
   })
 
   const [isOwner, setIsOwner] = useState(false)
 
+  const [updatedUser] = useMutation(UPDATE_USER)
+
+  let onCardClickHandler = async (e, card) => {
+    setSelectedCard(card)
```
Comment (DaceyTom2) [inline pages/room/[id].js:46]: Not directly related to the PR but I'd consider removing setSelectedCard and replace it with matching on users for current card selected. Also, the currPlayer logic to just point to your own user in users

**Review by DaceyTom2 — APPROVED** [review]
Looks great, small comment that we should consider but most likely for a separate PR

File: pages/room/[id].js [hunk 2]
```diff
@@ -131,7 +156,7 @@ export default function Room(props) {
                   src={card.src}
                   id={card.id}
                   alt={card.alt}
-                  onClick={() => setSelectedCard(card)}
+                  onClick={(e) => onCardClickHandler(e, card)}
                   onKeyDown={(e) => {
```
Comment (nibivi77) [inline pages/room/[id].js:160]: `onKeyDown` should also call `onCardClickHandler`

File: pages/room/[id].js [hunk 3]
```diff
@@ -32,11 +34,35 @@ export default function Room(props) {
...
-    card: selectedCard,
+    card: null,
+    room: props.roomID,
   })
 
   const [isOwner, setIsOwner] = useState(false)
 
+  const [updatedUser] = useMutation(UPDATE_USER)
+
+  let onCardClickHandler = async (e, card) => {
```
Comment (nibivi77) [inline pages/room/[id].js:45]: Can we use `const` instead of let, please?

**Review by nibivi77 — APPROVED** [review]
Good work!

**Pull request merged**